src/app/component/user/components/profile/calendar/calendar-week/calendar-week.component.ts

```diff
@@ -0,0 +1,174 @@
+    this.isCurrentDayActive();
+  }
+
+  public calculateCalendarModel(day: number, month: number, year: number): void {
+    this.calendar.month = month;
+    this.calendar.year = year;
+    this.calendar.firstDay = (new Date(year, month, 0)).getDay();
+    this.calendar.lastDayInWeek = (this.calendar.firstDay + this.maxDaysInWeek);
+    this.calendar.totalDaysInMonth = this.getDaysInMonth(month, year);
+    this.monthAndYearName = `${this.currentDay} - ${this.currentDay + this.maxDaysInWeek}
+${this.months[this.currentMonth]} ${this.currentYear}`;
+  }
+
+  public bindCalendarModel(): void {
+    const end = 7;
+    const calendarDays = Array.from({ length: end },
+      (_, i) =>
+        this.getWeekTemplate(i + this.currentDay));
+    this.calendarDay = [...this.calendarDay, ...calendarDays];
+  }
+
+  public getWeekTemplate(days?: number): CalendarWeekInterface {
+    return {
+      numberOfDate : days || '',
+      date : new Date(),
+      month : this.calendar.month,
+      year : this.calendar.year,
+      firstDay : this.calendar.firstDay,
+      lastDayInWeek: this.calendar.lastDayInWeek,
+      totalDaysInMonth : this.calendar.totalDaysInMonth,
+      dayName : (new Date(this.calendar.year, this.calendar.month, days)
+        .toDateString()
+        .substring(0, 3)) || '',
+      isHabitsTracked: false,
+      isCurrentDayActive: false
+    };
+  }
+
+  public isCurrentDayActive(): void {
+    this.calendarDay.map(el => el.isCurrentDayActive =
+      (el.date.getDate() === el.numberOfDate
+        && el.date.getMonth() === el.month
+        && el.date.getFullYear() === el.year)
+    );
+  }
+
+  public markCurrentDayOfWeek(): void {
+    const option = { weekday: 'short' };
+    this.language = this.languageService.getCurrentLanguage();
+    this.calendarDay.find(el => {
+      if (el.isCurrentDayActive &&
+        el.date.getMonth() === el.month &&
+        el.date.getFullYear() === el.year) {
+        const dayName = (new Date(el.year, el.month, +el.numberOfDate)
+          .toLocaleDateString(this.language, option));
+        this.currentDayName = dayName.charAt(0).toUpperCase() + dayName.slice(1);
+      }
+    });
+  }
+
+  public nextWeek(): void {
+    if (this.currentDay === this.calendar.totalDaysInMonth - this.maxDaysInWeek) {
+      this.currentMonth = this.currentMonth + 1;
+      this.currentDay = 1;
+      const [firstDay, ...lastDays] = this.daysName;
+      this.daysName = [...lastDays, firstDay];
+    } else {
+      const [firstDay, ...lastDays] = this.daysName;
+      this.daysName = [...lastDays, firstDay];
+    }
+    this.calendarDay = [];
+    this.buildCalendar();
+  }
+
+  public previousWeek(): void {
+    if (this.currentDay === 1) {
+      this.currentMonth = this.currentMonth - 1;
+      this.currentDay = this.calendar.totalDaysInMonth - this.maxDaysInWeek;
+    } else {
+      this.currentDay = this.currentDay - 1;
+      this.daysName = [this.daysName[6], ...this.daysName.slice(0, 6)];
```

Comment:
the `slice` method returns new instance of array, so - you shouldn't use `spread` in such cases